homeassistant/components/wilight/switch.py

```diff
@@ -0,0 +1,352 @@
+ICON_PAUSE = "mdi:pause-circle-outline"
+
+
+def entities_from_discovered_wilight(api_device: Any) -> tuple[Any]:
+    """Parse configuration and add WiLight switch entities."""
+    entities: Any = []
+    for item in api_device.items:
+        if item["type"] == ITEM_SWITCH:
+            index = item["index"]
+            item_name = item["name"]
+            if item["sub_type"] == SWITCH_VALVE:
+                entities.append(WiLightValveSwitch(api_device, index, item_name))
+            elif item["sub_type"] == SWITCH_PAUSE_VALVE:
+                entities.append(WiLightValvePauseSwitch(api_device, index, item_name))
+
+    return entities
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up WiLight switches from a config entry."""
+    parent = hass.data[DOMAIN][entry.entry_id]
+
+    # Handle a discovered WiLight device.
+    entities = entities_from_discovered_wilight(parent.api)
+    async_add_entities(entities)
+
+    # Handle services for a discovered WiLight device.
+    async def set_watering_time(service: Any) -> None:
+        entity_id = service.data[ATTR_ENTITY_ID]
+
+        for entity in entities:
+            if (entity.entity_id == entity_id) & isinstance(entity, WiLightValveSwitch):
+                watering_time = service.data[ATTR_WATERING_TIME]
+                await entity.async_set_switch_time(watering_time=watering_time)
+
+    async def set_trigger(service: Any) -> None:
+        entity_id = service.data[ATTR_ENTITY_ID]
+
+        for entity in entities:
+            if (entity.entity_id == entity_id) & isinstance(entity, WiLightValveSwitch):
+                trigger_index = service.data[ATTR_TRIGGER_INDEX]
+                trigger = service.data[ATTR_TRIGGER]
+                await entity.async_set_trigger(
+                    trigger_index=trigger_index, trigger=trigger
+                )
+
+    async def set_pause_time(service: Any) -> None:
+        entity_id = service.data[ATTR_ENTITY_ID]
+
+        for entity in entities:
+            if (entity.entity_id == entity_id) & isinstance(
+                entity, WiLightValvePauseSwitch
+            ):
+                pause_time = service.data[ATTR_PAUSE_TIME]
+                await entity.async_set_switch_time(pause_time=pause_time)
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_SET_WATERING_TIME,
+        set_watering_time,
+        schema=SERVICE_SCHEMA_WATERING_TIME,
+    )
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_SET_TRIGGER,
+        set_trigger,
+        schema=SERVICE_SCHEMA_TRIGGER,
+    )
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_SET_PAUSE_TIME,
+        set_pause_time,
+        schema=SERVICE_SCHEMA_PAUSE_TIME,
+    )
+
+
+def wilight_to_hass_pause_time(value: int) -> int:
+    """Convert wilight pause_time seconds to hass hour."""
+    return int(value / 3600)
```

Comment:
Maybe round instead?
```suggestion
    return round(value / 3600)
```